### PR TITLE
Re-add hls-stan-plugin

### PR DESCRIPTION
**Pull request by 0rphee**
Hi! now that stan supports ghc-9.6 (https://github.com/kowainik/stan/pull/523), I figured that it would be possible to add its hls plugin back.

I tested it locally with 9.6.3.

9.8.1 support for the stan plugin tests requires to add `th-abstraction` to `allow-newer` in `cabal.project`, and depends on the publishing of 9.8.1 support to hackage for `hiedb`. (They are indirect dependencies, I'm not sure which dependency might trigger them, I'm not used to cabal)

If something more is needed, please tell me and I'll see what I can do :)

img

### CODE REVIEW

File: cabal.project
Comment (0rphee) [inline cabal.project:38]: I commented the index-state to get the new stan version.

Comment (fendor) [reply cabal.project:38]: You should just update the index-state. No point in commenting it out and the PR can't be merged without it any way.

**Review by fendor — COMMENTED** [review]
While it is great to have stan back, perhaps even with new diagnostics, we lack a maintainer for the plugin. Are you volunteering to maintain the stan plugin in the foreseeable future? If so, please add yourself to the `CODEOWNERS` file.

Since we do experience some issues with plugins that bitrot because it has no active maintainer, I don't think we will merge this until we have found a maintainer.

**Comment by 0rphee** [comment]
Maybe i could do it. What is expected of a maintainer? I suppose mainly keep it running with ghc changes and fix bug reports?

**Comment by fendor** [comment]
Exactly, and you will mainly be pinged in the case of hls-stan-plugin related issues, but it does not include fixing the bug upstream in stan.

**Comment by 0rphee** [comment]
Ok. I will mantain the plugin, I've added myself to the `CODEOWNERS` file.

**Review by fendor — APPROVED** [review]
LGTM, thank you very much for becoming the maintainer of the plugin!
Once the PR is approved, we will invite you to the repo as a contributor, then the Codeowners file will not display any errors any more.

File: docs/support/plugin-support.md
Comment (0rphee) [reply docs/support/plugin-support.md:65]: whoops! I forgot to change that line to `hls-stan-plugin`. Should I change it or completely remove it?

Comment (fendor) [reply docs/support/plugin-support.md:65]: Change it, please, we still want it for documentation :)

**Comment by michaelpj** [comment]
I wonder if @tomjaguarpaw would also be interested in being involved?

**Comment by tomjaguarpaw** [comment]
Hi, thanks for pinging me. I don't have the bandwidth to be a maintainer of the plugin myself, but I'm happy to unblock any issues that need help on the upstream `stan` side.

**Comment by fendor** [comment]
@0rphee You received an invite to this repo! Once you have accepted, we will merge this PR :) Thank you for your patience!

**Comment by michaelpj** [comment]
(and please fix the conflicts)

**Comment by 0rphee** [comment]
I've accepted the invitation to the repo :)

The conflicts are fixed. From what I've seen, the stack lts21 previous build failure was caused by a now fixed issue (#3868), should I rebase it on the newest master?. Also, I now remembered about the hls-stan-plugin test suite (I forgot to add it to the gh actions workflow), should I still add it back?

**Comment by fendor** [comment]
> The conflicts are fixed. From what I've seen, the stack lts21 previous build failure was caused by a now fixed issue (https://github.com/haskell/haskell-language-server/pull/3868). 

:confetti_ball: 

> Also, I now remembered about the hls-stan-plugin test suite (I forgot to add it to the gh actions workflow), should I still add it back?

Ah, yes please, add it back to the GitHub actions. Otherwise, we won't run the test suite in CI!

**Comment by 0rphee** [comment]
I think everything is fine now!

**Review by fendor — APPROVED** [review]
LGTM!

**Comment by tomjaguarpaw** [comment]
This is great to see! Thanks @0rphee.

**Pull request merged**
